### PR TITLE
`azurerm_function_app` - add support for the `always_on` and `connection_string` fields

### LINKED ISSUE #691: Feature request: add alwaysOn and connectionStrings arguments in azurerm_function_app
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
any

### Affected Resource(s)
Please list the resources as a list, for example:
- `azurerm_function_app`

### Expected Behavior
The `azurerm_function_app` resource should accept the `alwaysOn` and the `connectionStrings`  arguments (similar to the `azurerm_app_service` resource).

### Actual Behavior
Only the `app_settings` argument is currently accepted.

### PR DESCRIPTION
**Pull request by cloudify**
Hello @tombuildsstuff,

my attempt to fix #691, I've added the `always_on` attribute for now, if my approach is good for you I'll add the rest,

thanks

### CODE REVIEW

**Comment by cloudify** [comment]
I've added also the support for custom `connection_string` in `function_app`

File: azurerm/resource_arm_function_app_test.go [hunk 1]
```diff
@@ -94,6 +94,52 @@ func TestAccAzureRMFunctionApp_appSettings(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMFunctionApp_siteConfig(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (cloudify) [reply azurerm/resource_arm_function_app_test.go]: done

File: azurerm/resource_arm_function_app.go [hunk 2]
```diff
@@ -106,6 +159,7 @@ func resourceArmFunctionAppCreate(d *schema.ResourceData, meta interface{}) erro
 			ServerFarmID: utils.String(appServicePlanID),
 			Enabled:      utils.Bool(enabled),
 			SiteConfig: &web.SiteConfig{
+				AlwaysOn:    siteConfig.AlwaysOn,
 				AppSettings: &basicAppSettings,
```
[the remark this answers is not in the document]
Comment (cloudify) [reply azurerm/resource_arm_function_app.go]: done

**Comment by cloudify** [comment]
@tombuildsstuff I've fixed the conflict and taken care of your suggestions

**Comment by tombuildsstuff** [comment]
Hey @cloudify 

Thanks for pushing those updates - apologies for the delayed re-review here, I'm just taking a look into this now.

Thanks!

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-01-23 at 13 09 54

**Pull request merged**